src/cli/commands/zone.rs

```diff
@@ -155,30 +155,107 @@ impl Zone {
             })?;
 
         match response {
-            Ok(status) => {
-                Self::print_zone_status(status);
-                Ok(())
-            }
+            Ok(status) => Self::print_zone_status(client, status).await,
             Err(ZoneStatusError::ZoneDoesNotExist) => {
                 println!("zone `{zone}` does not exist");
                 Err(())
             }
         }
     }
 
-    fn print_zone_status(zone: ZoneStatus) {
-        println!("{}", zone.name);
-        println!("  source: {}", zone.source);
-        println!("  policy: {}", zone.policy);
-        println!("  stage: {}", zone.stage);
+    async fn print_zone_status(client: CascadeApiClient, zone: ZoneStatus) -> Result<(), ()> {
+        let url = format!("policy/{}", zone.policy);
+        let response: Result<PolicyInfo, PolicyInfoError> = client
+            .get(&url)
+            .send()
+            .and_then(|r| r.json())
+            .await
+            .map_err(|e| {
+                error!("HTTP request failed: {e}");
+            })?;
+
+        let policy = match response {
+            Ok(policy) => policy,
+            Err(PolicyInfoError::PolicyDoesNotExist) => {
+                println!(
+                    "policy `{}` used by zone `{}` does not exist",
+                    zone.policy, zone.name
+                );
+                return Err(());
+            }
+        };
+
+        println!("Zone: {}", zone.name);
+        print!("Status: At the {} pipeline stage", zone.stage);
```

Comment:
My main gripe this format is that it feels very wordy instead of structured. Maybe that's the right call, but I don't know. When I was writing the previous version I tried to make it a clear key-value format (bit like YAML I suppose). This feels more descriptive but harder to glance over to extract the information you need if you're already familiar with it.

Finally, a small nit:
```suggestion
        print!("Status: At the {} pipeline stage ", zone.stage);
```

Reply:
One issue in particular I have with this "wordy" approach is it's unclear to the reader what the possible texts are that can appear on a given line and what fields that might contain (e.g. zonefile source vs XFR source).